.github/workflows/tekton_bundle_push.yaml

```diff
@@ -0,0 +1,93 @@
+  API_TOKEN: ${{ secrets.QUAY_API_TOKEN }}
+jobs:
+  run-pipeline:
+    name: Tekton Bundle
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2  # using git diff-tree requires fetch depth=2
+      - run: |
+          mkdir -vp $HOME/.kube || true
+          cat <<-EOF > $HOME/.kube/config
+          apiVersion: v1
+          kind: Config
+          clusters:
+          - cluster:
+              server: _
+            name: _
+          contexts:
+          - context:
+              cluster: _
+            name: _
+          current-context: _
+          EOF
+      - uses: jerop/tkn@v0.1.0
+      - uses: redhat-actions/podman-login@v1
+        with:
+          username: ${{ env.REGISTRY_USER }}
+          password: ${{ env.REGISTRY_PASSWORD }}
+          registry: ${{ env.IMAGE_REGISTRY }}
+      - run: |
+          declare -A bundle_dirs
+          while read file
+          do
+            [[ ${file%%/*} = \.* ]] && continue
+            bundle_dirs["${file%%/*}"]+="${file##*/} "
+          done < <(git diff-tree --no-commit-id --name-only -r ${{ github.sha }})
+          for BUNDLE_NAME in "${!bundle_dirs[@]}"
+          do
+            BUNDLE_FILES=()
+            for file in $BUNDLE_NAME/*.yaml
+            do
+              [[ -e "$file" ]] || continue
+              BUNDLE_FILES+=("$file")
+            done
+            printf -v bundle_files -- '-f %s ' ${BUNDLE_FILES[*]}
+            printf -v image_string '%s/%s/%s' \
+              ${{ env.IMAGE_REGISTRY }} \
+              ${{ env.IMAGE_NAMESPACE }} \
+              "${BUNDLE_NAME}"
+
+            API_HTTP="https://${{ env.IMAGE_REGISTRY }}/api/v1"
+
+            mapfile -t repos < <(
+              curl \
+                --silent \
+                --request GET "${API_HTTP}/repository?namespace=${{ env.IMAGE_NAMESPACE }}" \
+                --header 'Authorization: Bearer ${{ secrets.QUAY_API_TOKEN }}' | \
+              jq '.repositories[].name'
+            )
+            if ! grep "${BUNDLE_NAME}" <<<${repos[*]}
+            then
+              # Need to create the repo
+              printf -v repo_string -- \
+                '{"namespace": "%s", "repository": "%s", "description": "%s", "visibility": "%s", "repo_kind": "%s"}' \
+                ${{ env.IMAGE_NAMESPACE }} \
+                ${BUNDLE_NAME} \
+                ${BUNDLE_NAME} \
+                public \
+                image
+              curl \
+                --silent \
+                --request POST "${API_HTTP}/repository?namespace=${{ env.IMAGE_NAMESPACE }}" \
+                --header 'Authorization: Bearer ${{ secrets.QUAY_API_TOKEN }}' \
+                --header 'Content-Type: application/json' \
+                --data "$repo_string"
+            fi
+
+            tkn bundle push ${bundle_files} "${image_string}:${GITHUB_SHA:0:8}"
+            tkn bundle push ${bundle_files} "${image_string}:latest"
```

Comment:
It's a bad idea to use `latest` tag. We can remove the line all together or change it by `stable` or `main`. I'd go for renaming the tag to `main` to state it contains a bundle with the code shown in the `main` branch.

Reply:
> Much better, although I haven't tested it. Trusting you here.
Trust but verify (read sarcastically) 
https://www.youtube.com/watch?v=As6y5eI01XE

Jokes aside, I'll be showing a demo at stand-up meeting, so you will get to see it working.

Meanwhile I'm going to switch to using "main" tag instead of "latest".
Actually, based on what you wrote, I love the idea of image tags following git branch or tag refs. So for example, say the branch is "feature-foo", then the quay.io image tag will follow. But for now only "main" branch is able to triggering this workflow, and so "main"  will probably be the only image tag you see from now forward.

In regards to scripts... I'd rather call a python script or write a GO package, something, anything besides crufty bash script. So yes, I love that idea. I'll have to think about how that will work, because I'll need access to GitHub environments variables and contexts.

Reply:
I don't think that using the feature branch name is a good idea. I, personally, don't think much on the quality of my branch name. Tags is a different story, but we don't want to maintain tags for individual bundles.

Regarding the script, bash is totally fine for me and it's already written. In fact, leaving the workflow as it is, is also fine. A different file would be better IMHO but it's not necessary. It would look something like this:
```
jobs:
  run_tests:
    runs-on: ubuntu-20.04
    steps:
    - uses: actions/checkout@v2
    - name: Run script file
      run: |
         ./scripts/build_and_push_bundles.sh
      shell: bash
```
Your env variables should be already accessible.

Also, please keep in mind the proposed project structure. Bundles will be in a deeper level (inside `definitions`).

Reply:
What about the TaskRef inside the Pipeline's?
For example:
```
taskRef:
        bundle: quay.io/hacbs-release/hello-world:0.3-alpine
        name: echo
```
The tag is currently hard coded.
Will that need to change to the digest tag, or can we make the assumption the symbolic name would always be "main"?

Reply:
I've pushed an update to move the bundles under the "definitions" dir, and some minor refactoring.
I think this is almost ready to merge.

I'd like to hold off on moving the workflow to a separate script for now, so I can get my Jira finished with the minimum acceptance critera.

Reply:
Going to move the workflow to a separate script later, resolving for now, will circle-back in the near-future.